### PR TITLE
Changes Werewolf Serum to mix based on moon phase

### PR DESCRIPTION
**Pull request by TerraGS**
## About the PR 

Updates werewolf serum from only reacting on full moon days to reacting on any day with the addition of a fail case based on probablility that scales with the current moon phase (i.e. will succeed 100% of the time on full moon, 0% on new moon).

## Why's this needed? 

Werewolf serum is a piece of content that rarely gets a chance for interaction as is other than occasionally making vampire serum from it. This will encourage engaging with it more often while not making it any easier to obtain. Since werewolf antag can happen any day I feel comfortable breaking away from the strict full moon requirement while keeping the flavor.

## Changelog

```changelog
(u)TerraGS
(*)Werewolf Serum now reacts based on how close to a full moon the date is.
```

### CODE REVIEW

**Comment by TDHooligan** [comment]
couldnt you just make minimum-size batches of it repeatedly until you got the result, but now on any day of the year?
it was made like this so giganerds couldnt use it every round of every day

**Comment by TerraGS** [comment]
> couldnt you just make minimum-size batches of it repeatedly until you got the result, but now on any day of the year? it was made like this so giganerds couldnt use it every round of every day

You present an interesting case. The probability of a single success out of multiple attempts is 1-(1-Prob)^Attempts, so assuming someone finds 20u of werewolf pt4 they'd get roughly 6 attempts at mixing a minimum dose and on a 50% chance day there would be a 98.4% of at least one dose. I could adjust my formula from linear to be exponential rather than linear to compensate for this as well as the edge case of players that attempt this to the maximum of their opportunity.

**Comment by TerraGS** [comment]
Additionally the current probability of imperfect serum spawning in an art can is 5%. Assuming an average round is 45 min and someone finds an art can every round, imperfect serum would have roughly 83% chance of being found across any given day. I say this in the interest of assurance that there is still significant barrier to abuse of this feature.

File: code/modules/chemistry/Chemistry-Recipes.dm [hunk 1]
```diff
@@ -3572,11 +3571,31 @@ datum
...
 
-			does_react(var/datum/reagents/holder)
-				return time2text(world.realtime, "MMM DD") in full_moon_days_2053 //just doesn't react unless it's a full moon
+			on_reaction(var/datum/reagents/holder, var/created_volume)
+				var/my_atom = holder.my_atom
+				if(!my_atom) return
+				var/turf/reaction_loc = get_turf(my_atom)
+				if(!reaction_loc) return
+				var/date = time2text(world.realtime, "MMM DD")
+				if(prob((moon_list_2053[date] ** 4) / 100))
```
Comment (Tarmunora) [inline code/modules/chemistry/Chemistry-Recipes.dm:3594]: I do not think this expression is correct. Check your math.

**Review by Tarmunora — CHANGES_REQUESTED** [review]

File: code/modules/chemistry/Chemistry-Recipes.dm (hunk 1 above)
Comment (TerraGS) [reply code/modules/chemistry/Chemistry-Recipes.dm]: Must have been really tired when I wrote that. Fixed.

**Comment by Tarmunora** [comment]
Hm. to be honest, I don't think that more non-antag werewolves is something that we want, at least not for the immediate future.

**Pull request closed without merging**